### PR TITLE
feat(platform-web): redirect support thirdy-party sites

### PR DESCRIPTION
**Pull request by wangjinyang**
**What kind of change does this PR introduce?**

feat(platform-web): redirect support thirdy-party sites

**Did you add tests(unit,e2e) for your changes?**

yes

**Does this PR introduce a breaking change?**

no

**What needs to be documented once your changes are merged?**

redirect support thirdy-party sites

### CODE REVIEW

File: packages/platform-web/src/shuvi-app/app/server.ts [hunk 1]
```diff
@@ -36,9 +42,18 @@ export const createApp: CreateAppServer = options => {
         app.setLoadersData(loaderResult);
       } catch (error: any) {
         if (isRedirect(error)) {
+          const location = error.headers.get('Location')!;
+          const isThirdSite = thirdSite(location);
+          const status = error.status;
           next({
-            path: error.headers.get('Location')!,
-            skipGuards: true
+            path: isThirdSite ? pathToString(to) : location,
```
Comment (liximomo) [inline packages/platform-web/src/shuvi-app/app/server.ts:49]: 这里直接to.pathname 就行了

File: packages/platform-web/src/shuvi-app/app/server.ts [hunk 2]
```diff
@@ -36,9 +42,18 @@ export const createApp: CreateAppServer = options => {
...
+            replace: true,
+            skipGuards: true,
+            state: {
+              isThirdSite,
```
Comment (liximomo) [inline packages/platform-web/src/shuvi-app/app/server.ts:53]: isThirdSite 不需要穿了吧，直接在这里判断一下  

{
 location: isThirdSite ? location : to.pathname
}

**Review by liximomo — CHANGES_REQUESTED** [review]

File: packages/platform-web/src/shuvi-app/app/server.ts (hunk 1 above)
Comment (wangjinyang) [reply packages/platform-web/src/shuvi-app/app/server.ts:49]: to上不仅有pathname，还可能有query和hash参数

File: packages/platform-web/src/shuvi-app/app/server.ts (hunk 2 above)
Comment (wangjinyang) [reply packages/platform-web/src/shuvi-app/app/server.ts:53]: 传递相对路径，to要通过route内部，使用当前的路径作为base解析成绝对路径。

to上不仅有pathname，还可能有query和hash参数， 重新向也需要考虑

File: packages/platform-web/src/shuvi-app/react/view/ReactView.server.tsx [hunk 3]
```diff
@@ -19,14 +19,22 @@ export class ReactServerView implements IReactServerView {
...
     }
 
     if (redirected) {
-      return redirect(pathname);
+      const { location, status, isThirdSite } = state as {
+        isThirdSite: boolean;
+        location: string;
+        status: number;
+      };
+      return redirect(
```
Comment (liximomo) [inline packages/platform-web/src/shuvi-app/react/view/ReactView.server.tsx:34]: router.resolve(to, from)

**Review by liximomo — CHANGES_REQUESTED** [review]

**Pull request merged**